### PR TITLE
Add and improve coverage in UT

### PR DESCRIPTION
**Pull request by nicklera1**
**Issue #:** 

**Description of changes:**
Add new UT as well increase coverage on some of the existing tests.
Screen Shot 2020-08-06 at 12 40 16 PM
**Testing**
1. Have you successfully run `npm run build:release` locally? yes
#### BEFORE
```
=============================== Coverage summary ===============================
Statements   : 88.75% ( 1388/1564 )
Branches     : 76.61% ( 501/654 )
Functions    : 80.41% ( 357/444 )
Lines        : 90.78% ( 1211/1334 )
================================================================================

Test Suites: 43 passed, 43 total
Tests:       169 passed, 169 total
Snapshots:   0 total
Time:        95.135s
Ran all test suites.
```
#### AFTER 
```
=============================== Coverage summary ===============================
Statements   : 92.46% ( 1446/1564 )
Branches     : 80.43% ( 526/654 )
Functions    : 84.68% ( 376/444 )
Lines        : 94% ( 1254/1334 )
================================================================================

Test Suites: 45 passed, 45 total
Tests:       178 passed, 178 total
Snapshots:   0 total
Time:        187.285s
```
2. How did you test these changes?
yes
3. If you made changes to the component library, have you provided corresponding documentation changes?
n/a
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tst/components/ui/Roster/index.test.tsx [hunk 1]
```diff
@@ -5,8 +5,32 @@ import '@testing-library/jest-dom';
...
+          onSearch={() => {}}
+        />
+      </Roster>
+    );
+    const { getByLabelText } = renderWithTheme(lightTheme, component);
+
+    getByLabelText('Open search').click();
+    const searchLbl = getByLabelText('Search');
+    expect(searchLbl).toBeInTheDocument();
+    getByLabelText('clear Search').click();
```
Comment (ziyiz-amzn) [inline tst/components/ui/Roster/index.test.tsx:31]: It will be good to assert search button is not rendered before pressed, and it is rendered afterwards.

Comment (nicklera1) [reply tst/components/ui/Roster/index.test.tsx:31]: added!

File: tst/components/ui/Modal/ModalButtonGroup.test.tsx [hunk 2]
```diff
@@ -25,6 +25,39 @@ describe('ModalButtonGroup', () => {
     const el = getByTestId('modal-button-group');
     expect(el).toBeInTheDocument();
   });
+  it('renders a group with a Button & secondary Button', () => {
```
[the remark this answers is not in the document]
Comment (nicklera1) [reply tst/components/ui/Modal/ModalButtonGroup.test.tsx:28]: done

File: tst/components/ui/Modal/ModalButtonGroup.test.tsx [hunk 3]
```diff
@@ -12,34 +12,103 @@ import lightTheme from '../../../../src/theme/light';
...
+          secondaryButtons={[<ModalButton label={secondaryButtonLbl} />]}
+        />
+      </ModalContext.Provider>
+    );
+    const { getByTestId, getByLabelText } = renderWithTheme(
+      lightTheme,
+      component
+    );
+    const el = getByTestId('modal-button-group');
+    expect(el).toBeInTheDocument();
```
Comment (ziyiz-amzn) [inline tst/components/ui/Modal/ModalButtonGroup.test.tsx:52]: nit: this is asserted in the test case above, we may only assert what we aim to test in this particular case. Can remove these 2 lines.

File: tst/components/ui/Modal/ModalButtonGroup.test.tsx [hunk 4]
```diff
@@ -12,34 +12,103 @@ import lightTheme from '../../../../src/theme/light';
...
       </ModalContext.Provider>
     );
-    const { getByTestId } = renderWithTheme(lightTheme, component);
+    const {
+      getByTestId,
+      getByLabelText,
+      queryAllByLabelText
+    } = renderWithTheme(lightTheme, component);
     const el = getByTestId('modal-button-group');
     expect(el).toBeInTheDocument();
```
Comment (ziyiz-amzn) [inline tst/components/ui/Modal/ModalButtonGroup.test.tsx:72]: The same.

**Pull request merged**